pkg/chains/storage/oci/legacy.go

```diff
@@ -119,12 +119,22 @@ func (b *Backend) uploadSignature(ctx context.Context, format simple.SimpleConta
 	imageName := format.ImageName()
 	logger.Infof("Uploading %s signature", imageName)
 
-	ref, err := newDigest(b.cfg, imageName)
+	var opts []name.Option
+	if b.cfg.Storage.OCI.Insecure {
+		opts = append(opts, name.Insecure)
+	}
+
+	ref, err := name.NewDigest(imageName, opts...)
```

Comment:
I don't think we need the insecure option when calling `NewDigest`. That is just doing a simple string parsing and not actually hitting the registry.